pkg/channel/http/http_channel.go

```diff
@@ -222,8 +221,8 @@ func (h *Channel) constructRequest(ctx context.Context, req *invokev1.InvokeMeth
 
 	// HTTP client needs to inject traceparent header for proper tracing stack.
 	span := diag_utils.SpanFromContext(ctx)
-	httpFormat := &tracecontext.HTTPFormat{}
-	tp, ts := httpFormat.SpanContextToHeaders(span.SpanContext())
+	tp := diag.SpanContextToW3CString((*span).SpanContext())
```

Comment:
Why are these `(*span)` necessary? They look... awkward. Could `diag_utils.SpanFromContext` return a pointer directly?

Reply:
Sure. Just for my knowledge: Is that the idiomatic way rather than create a struct to wrap the interface? Thanks

Reply:
Yes, structs implement interfaces rather than wrapping them, usually. Take a look at this which seems to be well-written: https://medium.com/rungo/interfaces-in-go-ab1601159b3a 

An example is a readable stream.

```go
buf := bytes.Buffer{}
```

`buf`, whose type is `bytes.Buffer` implements some interfaces, including for example `io.Reader` (both `bytes.Buffer` and `*bytes.Buffer` would implement that interface).

So if you have a variable of type `io.Reader` or a method that accepts an `io.Reader`, you can pass that directly:

```go
// Assign buf to a variable of type io.Reader
var myreader io.Reader
myreader = buf

// Pass buf to a method that accepts an io.Reader
io.ReadAll(buf)
```

As for handling nils, variables that are of an interface type can always be nil, even when the type you're assigning to it is not a pointer.

```go
if myreader != nil { /* ... */ }
```

Hope this (and the article above) help :)

Although what you wrote is probably ok, the fact that it uses an uncommon syntax can be hard to follow and maintain in the future IMHO

Reply:
Ping @juazasan

Reply:
Thank @ItalyPaleAle for the example. However not sure they are representative of the scenario here. In your example bytes.Buffer is a struct that implements an interface and that is what you pass around, while the challenge here is that the Span creator from Opentelemetry returns objects of type interface not a struct. See: https://pkg.go.dev/go.opentelemetry.io/otel/trace#Tracer

Reply:
Anyways, I will look at changing the functions signatures and pass the variables of type interface by value instead of by reference as you are recommending instead of creating my own struct that wraps the object returned by the creator functions.

Reply:
@juazasan I have to admit that I'm a bit confused as to where the issue is in this case. Where in the code are you defining the struct that you need to pass around?

Reply:
I have not defined it for now. I would have defined it for instance in the diagnostic_utils package if we had agreed on that being the best option to get rid of the (*.span).

Reply:
@ItalyPaleAle This issue should have been addressed with my last commit.